ocrd_network/ocrd_network/utils.py

```diff
@@ -110,3 +111,59 @@ def post_to_callback_url(logger, callback_url: str, result_message: OcrdResultMe
     }
     response = requests.post(url=callback_url, headers=headers, json=json_data)
     logger.info(f'Response from callback_url "{response}"')
+
+
+def get_ocrd_workspace_instance(mets_path: str, mets_server_url: str = None) -> Workspace:
+    if mets_server_url:
+        if not is_mets_server_running(mets_server_url=mets_server_url):
+            raise RuntimeError(f'The mets server is not running: {mets_server_url}')
+
+    workspace = Resolver().workspace_from_url(
+        mets_url=mets_path,
+        mets_server_url=mets_server_url
+    )
+    return workspace
+
+
+def get_ocrd_workspace_physical_pages(mets_path: str, mets_server_url: str = None) -> List[str]:
+    physical_pages = get_ocrd_workspace_instance(
+        mets_path=mets_path,
+        mets_server_url=mets_server_url
+    ).mets.physical_pages
+
+    # This conversion is required since
+    # the type of physical pages is not str
+    str_physical_pages = []
+    for physical_page in physical_pages:
+        str_physical_pages.append(f'{physical_page}')
```

Comment:
Why is that conversion necessary, `physical_pages` should return a list of `str`?

Reply:
It does not. I spent some time debugging that. The `physical_pages` returns a list of XML tree elements. Then when the `OcrdProcessingMessage` is encoded to bytes by the Processing Server, and decoded from bytes by the Processing Worker, it creates an XML tree element object, not `str`.

Reply:
This is strange and surprising. Can you share the workspace where this happens? We do explicitly test this behavior in the `tests/model/test_ocrd_mets.py`:

```python
def test_physical_pages(sbb_sample_01):                               
    assert len(sbb_sample_01.physical_pages) == 3, '3 physical pages' 
    assert isinstance(sbb_sample_01.physical_pages, list)             
    assert isinstance(sbb_sample_01.physical_pages[0], str)           
```

If the return is not a `List[str]`, this is a bug and needs fixing.

Reply:
It happens with any workspace I have tried. Check it for yourself, by running:

```python
from ocrd import Resolver, Workspace

METS_PATH = '/path/to/mets'
physical_pages = Resolver().workspace_from_url(mets_url=METS_PATH).mets.physical_pages

print(physical_pages[0])
print(type(physical_pages[0]))
print(type(f'{physical_pages[0]}'))
```

Which prints:
```bash
PHYS_0001
<class 'lxml.etree._ElementUnicodeResult'>
<class 'str'>
```

Reply:
I think this happens because `lxml.etree._ElementUnicoreResults` extends the `str` which the `isinstance` sees as a string, but it is not.

Reply:
I see, thanks, that explains it, I'll prepare a quick fix for that.

Reply:
https://github.com/OCR-D/core/pull/1102

Reply:
Approved.